postprocess/postprocess.go

```diff
@@ -0,0 +1,231 @@
+// Package postprocess contains functionality to postprocess pointcloud maps
+package postprocess
+
+import (
+	"bytes"
+	"errors"
+	"image/color"
+	"math"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/pointcloud"
+)
+
+// Instruction describes the action of the postprocess step.
+type Instruction int
+
+const (
+	// Add is the instruction for adding points.
+	Add Instruction = iota
+	// Remove is the instruction for removing points.
+	Remove = iota
+)
+
+const (
+	fullConfidence = 100
+	removalRadius  = 100 // mm
+	xKey           = "X"
+	yKey           = "Y"
+
+	// ToggleCommand can be used to turn postprocessing on and off.
+	ToggleCommand = "postprocess_toggle"
+	// AddCommand can be used to add points to the pointcloud  map.
+	AddCommand = "postprocess_add"
+	// RemoveCommand can be used to remove points from the pointcloud  map.
+	RemoveCommand = "postprocess_remove"
+	// UndoCommand can be used to undo last postprocessing step.
+	UndoCommand = "postprocess_undo"
+)
+
+var (
+	// ErrPointsNotASlice denotes that the points have not been properly formatted as a slice.
+	ErrPointsNotASlice = errors.New("could not parse provided points as a slice")
+
+	// ErrPointNotAMap denotes that a point has not been properly formatted as a map.
+	ErrPointNotAMap = errors.New("could not parse provided point as a map")
+
+	// ErrXNotProvided denotes that an X value was not provided.
+	ErrXNotProvided = errors.New("could X not provided")
+
+	// ErrXNotFloat64 denotes that an X value is not a float64.
+	ErrXNotFloat64 = errors.New("could not parse provided X as a float64")
+
+	// ErrYNotProvided denotes that a Y value was not provided.
+	ErrYNotProvided = errors.New("could X not provided")
+
+	// ErrYNotFloat64 denotes that an Y value is not a float64.
+	ErrYNotFloat64 = errors.New("could not parse provided X as a float64")
+
+	// ErrRemovingPoints denotes that something unexpected happened during removal.
+	ErrRemovingPoints = errors.New("unexpected number of points after removal")
+)
+
+// Task can be used to construct a postprocessing step.
+type Task struct {
+	Instruction Instruction
+	Points      []r3.Vector
+}
+
+// ParseDoCommand parses postprocessing DoCommands into Tasks.
+func ParseDoCommand(
+	unstructuredPoints interface{},
+	instruction Instruction,
+) (Task, error) {
+	pointSlice, ok := unstructuredPoints.([]interface{})
+	if !ok {
+		return Task{}, ErrPointsNotASlice
+	}
+
+	task := Task{}
+	for _, point := range pointSlice {
+		pointMap, ok := point.(map[string]interface{})
+		if !ok {
+			return Task{}, ErrPointNotAMap
+		}
+
+		x, ok := pointMap[xKey]
```

Comment:
I think you could
```suggestion
		x, ok := pointMap[xKey].(float64)
```
To make it briefer